cql3/functions/aggregate_function.hh

```diff
@@ -56,6 +63,12 @@ public:
          */
         virtual opt_bytes compute(cql_serialization_format sf) = 0;
 
+        virtual void set_accumulator(const opt_bytes& acc) = 0;
+
+        virtual opt_bytes get_accumulator() const = 0;
+
+        virtual void reduce(cql_serialization_format sf, const opt_bytes& acc) = 0;
+
```

Comment:
I think we should move out accumulator storage out of aggregate functions, so we don't need to call new_aggregate. Functions should be pure const functions without state. That's out of scope for this series of course, but would probably have made it simpler if we started with it.